x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealmSingleNodeTests.java

```diff
@@ -110,10 +121,72 @@ protected Settings nodeSettings() {
         return builder.build();
     }
 
+    @Override
+    protected String configRoles() {
+        return super.configRoles() + "\n" + """
+            anonymous:
+              cluster:
+                - monitor
+            """;
+    }
+
     protected boolean addMockHttpTransport() {
         return false;
     }
 
+    @SuppressWarnings("unchecked")
+    public void testInvalidJWTDoesNotFallbackToAnonymousAccess() throws Exception {
+        // anonymous access works when no valid Bearer
+        {
+            Request request = new Request("GET", "/_security/_authenticate");
+            RequestOptions.Builder options = RequestOptions.DEFAULT.toBuilder();
+            // "Bearer" token missing or blank
+            if (randomBoolean()) {
+                options.addHeader("Authorization", "Bearer    ");
```

Comment:
nit: I think there might be a tiny difference w.r.t. the elder behavior. I haven't fully tested but I think the elder behavior would only allow anonymous access for `Bearer ` but not `Bearer  ` (note one vs. multiple spaces). I could be wrong and it probably doesn't matter and it was probably was never the intention to be white space dependent. I could also be totally wrong, but thought it worth a mention in case you think it matters. (I pulled `header.regionMatches(true, 0, "Bearer ", 0, "Bearer ".length()) && header.length() > "Bearer ".length()` into my IDE to get a better sense of exactly what that does and is the basis of this comment)

Reply:
> nit: I think there might be a tiny difference w.r.t. the elder behavior. I haven't fully tested but I think the elder behavior would only allow anonymous access for Bearer but not Bearer

I think the behavior, whatever that is, stays unchanged from this PR.
The method to extract the bearer token hasn't changed in this PR (the redundant one from `AuthenticatorChain#shouldFallbackToAnonymous` was identical to the reused `Authenticator#extractBearerTokenFromHeader`).

After debugging, it looks like both `Bearer ` and `Bearer.  ` get truncated to `Bearer` in the netty http decoder (somewhere, havent't pinpointed), which means that ES (both before and after this PR) interprets it as if there's no bearer credential present.